pyccel/parser/semantic.py

```diff
@@ -1466,6 +1467,10 @@ def format_shape(s):
                 new_expressions.append(Allocate(var,
                     shape=d_var['shape'], order=d_var['order'],
                     status=status))
+            elif isinstance(var.dtype, CustomDataType):
+                if var in self._allocs[-1]:
+                    self._allocs[-1].remove(var)
```

Comment:
Yeah because in this case it will be duplicated because of https://github.com/pyccel/pyccel/blob/f723c5dca2612a4465d54b9850b0ecf72e3fd15b/pyccel/parser/semantic.py#L2505 so I remove it from the self._allocs.